testing/tests/DevExpress.ui.widgets.form/form.tests.js

```diff
@@ -4032,6 +4032,75 @@ function getColsCountFromDOM($form) {
     return result;
 }
 
+QUnit.test('group colCountByScreen property change should update layout', function(assert) {
+    const form = $('#form').dxForm({
+        screenByWidth: () => 'md',
+        items: [{
+            itemType: 'group',
+            colCountByScreen: { md: 1 },
+            items: ['name', 'email']
+        }],
+    }).dxForm('instance');
+
+    form.option('items[0].colCountByScreen.md', 2);
+
+    assert.strictEqual(getColsCountFromDOM($(form.$element())), 2);
+});
+
+QUnit.test('nested group colCountByScreen property change should update layout', function(assert) {
+    const form = $('#form').dxForm({
+        screenByWidth: () => 'md',
+        items: [{
+            itemType: 'group',
+            items: [{
+                itemType: 'group',
+                colCountByScreen: { md: 2 },
+                items: ['Phone', 'Email'],
+            }],
+        }],
+    }).dxForm('instance');
+
+    form.option('items[0].items[0].colCountByScreen.md', 1);
+
+    assert.strictEqual(getColsCountFromDOM($(form.$element())), 1);
```

Comment:
$(form.$element()) - additional wrapping is not required in this case. But you say it doesn't work for me without it. With this wrapper, you probably overcome the problem that hasClass() jquery - iterate over all elements and check against each of the array. There is a suspicion in the renderer that it does not go through all the elements, but looks only at the first one.

Let's try remove the wrapper and rewrite this piece so that it works well in both cases

  [1, 2, 3, 4].forEach(colCount => {
      if($lastCol.hasClass(`dx-col-${colCount - 1}`)) {
          result = colCount;
      }
  });

Let me know if I missed something.

Reply:
good catch, yes our jquery wrapper's 'hasClass' method seems inconsistent with jquery's hasClass